### PR TITLE
EDSC-2197: Submit projects with ECHO Order access method

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: static/src/js/util/accessMethods.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Determine if the selected access method for a give project collection is valid
+ * @param {Object} projectCollection Project collection config object
+ */
+export const isAccessMethodValid = (projectCollection) => {
```
Comment (abbottry) [inline static/src/js/util/accessMethods.js]: I'm a little confused what a project collection is, is this an object/class or just a payload that matches what used to be a project collection?

File: static/src/js/util/isProjectCollectionValid.js [hunk 2]
```diff
@@ -6,9 +6,16 @@ import { isEmpty } from 'lodash'
...
   // Here is where we can check the method to see if its valid
-  if (!isEmpty(method)) return true
-  return false
+  const [methodKey] = Object.keys(method)
+
+  if (methodKey === 'download' && !isEmpty(method)) return true
+
+  const accessMethod = method[methodKey]
+  const { isValid = false } = accessMethod || {}
+  // if (!isValid) return true
```
Comment (abbottry) [inline static/src/js/util/isProjectCollectionValid.js]: remove

File: static/src/js/components/AccessMethod/AccessMethod.js [hunk 3]
```diff
@@ -16,138 +120,50 @@ import { getValueForTag, hasTag } from '../../util/tags'
...
-          <i className="access-method__radio-tooltip fa fa-info-circle" />
-        </OverlayTrigger>
-      </Radio>
-    )
-  }
+    switch (type) {
+      case 'download':
+        radioList.push(downloadButton(collectionId))
+        break
+      case 'ECHO_ORDER':
```
[the remark this answers is not in the document]
Comment (macrouch) [reply static/src/js/components/AccessMethod/AccessMethod.js]: `download` and `ECHO_ORDER` is what the type needs to be for submission, right? I'm not sure what that value is for esi and opendap yet, those are mostly fillers for now

File: static/src/js/util/accessMethods.js (hunk 1 above)
Comment (macrouch) [reply static/src/js/util/accessMethods.js]: Its the config out of the store, I'll add that to the comment

File: static/src/js/components/AccessMethod/AccessMethod.js [hunk 4]
```diff
@@ -161,32 +177,48 @@ export const AccessMethod = ({
...
+            <EchoForm
+              collectionId={collectionId}
+              form={form}
+              methodKey={selectedAccessMethod}
+              rawModel={rawModel}
+              onUpdateAccessMethod={onUpdateAccessMethod}
+            />
+          )
+        }
+        {/* This is weird */}
```
[the remark this answers is not in the document]
Comment (macrouch) [reply static/src/js/components/AccessMethod/AccessMethod.js]: Ah, I wanted to bring this up with @trevorlang in a review

File: serverless/src/getAccessMethods/handler.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    if (optionDefinitions) {
+      const forms = await getOptionDefinitions(optionDefinitions, jwtToken)
+
+      forms.forEach((form) => {
+        const [key] = Object.keys(form)
+        accessMethods[key] = {
+          ...echoOrderData,
+          ...form[key],
+          type: 'ECHO_ORDER'
```
Comment (abbottry) [inline serverless/src/getAccessMethods/handler.js]: This shouldn't be hard coded, it should be the `echoOrderData.Type`, it should contain the `Type` from UMM S which in this case is `ECHO ORDERS` and in the other two will be `ESI` and `OPeNDAP` (See [here](https://github.com/nasa/Common-Metadata-Repository/blob/ea2c519d9a3647c1952da02c95d9a93f731ec7a2/umm-spec-lib/resources/json-schemas/service/umm/v1.2/umm-s-json-schema.json#L342))

**Pull request merged**